### PR TITLE
[Feature] Add dgl.reorder() to re-order graph according to specified …

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Description

A Utility API named dgl.reorder() is added into //python/dgl/transform.py which gives users a hint and interface that re-ordering graph before hand could obtain acceleration in train.

The real obtained speed-up varies according to the characteristics of dataset and the strategy applied. Dedicated research and measure is highly preferable.

Related issue ticket: https://github.com/dmlc/dgl/issues/3059
 
## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR
- [x] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by jermainewang** [comment]
Look like the history is messed up with some old changes.

**Comment by Rhett-Ying** [comment]
> Look like the history is messed up with some old changes.

yes. fixed now.

BTW, I failed to add reviewer for this PR? need membership privilege?

File: tests/compute/test_transform.py [hunk 1]
```diff
@@ -1480,5 +1480,86 @@ def test_remove_selfloop(idtype):
         raise_error = True
     assert raise_error
 
+@parametrize_dtype
+def test_reorder(idtype):
```
Comment (hetong007) [inline tests/compute/test_transform.py:1484]: This test is not added to `__main__` yet.

Comment (Rhett-Ying) [reply tests/compute/test_transform.py:1484]: Talked to Minjie. we test with pytest for these tests.

ubuntu@ip-172-31-24-140:~/dgl1/tests$ pytest-3 compute/test_transform.py::test_reorder
=================================== test session starts ===================================
platform linux -- Python 3.6.9, pytest-3.3.2, py-1.5.2, pluggy-0.6.0
rootdir: /home/ubuntu/dgl1/tests, inifile:
collected 2 items                                                                         

compute/test_transform.py ..                                                        [100%]

================================ 2 passed in 0.92 seconds =================================

**Comment by hetong007** [comment]
Could we work in a way such that the content of each commit appears properly, other than force-push?

**Comment by Rhett-Ying** [comment]
sure. I used to git commit --amend for fix and always keeps one commit only for one ticket.

**Comment by hetong007** [comment]
@Rhett-Ying Thanks. We squash and merge every PR by default thus no need to worry about the history in one PR/branch. Having multiple commits makes a PR easier to read and track the changes, especially when it is complex.

**Review by hetong007 — APPROVED** [review]

**Pull request merged**